### ISSUE TITLE
Custom keybinding for exiting fullscreen mode on excalidraw

### ISSUE THREAD

**Issue by mateusfbsoares** [zsviczian/obsidian-excalidraw-plugin#694]
**Is your feature request related to a problem? Please describe.**
I'm always frustrated when I'm on fullscreen mode on obsidian-excalidraw and hit the ESC key to stop writing and go to the Selection tool, but instead, the fullscreen mode is exited. I have then to hit ESC two times to go out of writing mode and in the process I go out of fullscreen mode, which I don't want. First gif demonstrates this.
This is also a problem when I'm previewing a markdown file inside excalidraw and I hit the ESC key (which is needed to use the VIM keybindings native to obsidian) and fullscreen mode also exits. Second gif demonstrates this

**Describe the solution you'd like**
I believe the ideal solution would be to have an option on the plugin options to be able to configure another keybiding for exiting fullscreen mode, instead of it being unchangeable, as it is now. This way I could configure something like "CTRL-F" and the ESC key would be free to function as intended on excalidraw.
This could be a toggle for gettin in and out of fullscreen mode.

**Describe alternatives you've considered**
The workaround for the first problem is to, instead of hting ESC to stop writing, just click with the mouse on an empy location, and it works similarly.
There are no alternatives for the vim mode problem. It makes impossible to use it as intended.

**Additional context**
GIF 1 shows how ESC works as a way of stop writing when you're not on fullscreen mode, and how it does not work when you are in fullscreen mode.
![gif1](https://user-images.githubusercontent.com/43099047/178513516-34125f6d-8874-4bbd-8a25-28cdad1408c6.gif)

GIF 2 shows how ESC is neede when on VIM Mode and how it conflicts with "exit fullscreen" behaviour when on fullscreen.
![gif2](https://user-images.githubusercontent.com/43099047/178513568-3fdfdfd5-b120-44d6-a81b-0b176bab19b4.gif)

**Comment by zsviczian** [comment]
Yes, this frustrates me as well sometimes. The exit fullscreen mode is not an Excalidraw keybinding, but an Electron Browser key binding. If you can find out how to change it, I am happy to apply the fix.

I am simply using the DOM element's requestFullscreen() function. 
https://github.com/zsviczian/obsidian-excalidraw-plugin/blob/b041088b597823803a3723c31d93482c3d13d18b/src/ExcalidrawView.ts#L610

**Comment by EaZyStudiio** [comment]
> Yes, this frustrates me as well sometimes. The exit fullscreen mode is not an Excalidraw keybinding, but an Electron Browser key binding. If you can find out how to change it, I am happy to apply the fix.
> 
> I am simply using the DOM element's requestFullscreen() function.
> 
> [obsidian-excalidraw-plugin/src/ExcalidrawView.ts](https://github.com/zsviczian/obsidian-excalidraw-plugin/blob/b041088b597823803a3723c31d93482c3d13d18b/src/ExcalidrawView.ts#L610)
> 
> Line 610 in [b041088](/zsviczian/obsidian-excalidraw-plugin/commit/b041088b597823803a3723c31d93482c3d13d18b)
> 
>  this.contentEl.requestFullscreen(); //{navigationUI: "hide"});

Actually it can be fixed I got this almost ingenious idea from [DonkeyPacific](https://github.com/DonkeyPacific/obsidian-full-screen-cross-platform-plugin), I mean the way they do full screen isn't what exactly we want but the idea kept me thinking and it caused a few issues just to fix my ESC key issue that come with the standard DOM element's requestFullscreen() fuction. worthy trade off i'll say 🤣🤣

I call it a simulated full screen mode, to get the "real" fullscreen mode like requestFullscreen() fuction it's still a pull request on [DonkeyPacific](https://github.com/DonkeyPacific/obsidian-full-screen-cross-platform-plugin) page.

Drawbacks, Native obsidian no plugins:
1. Does not work on multiwindow mode you know pop-out obsidian tabs/pages,
2. Vertical tabs/Stacked tabs, Honestly it just breaks it WORKS just not work work
3. 1 issue with Excalidraw specifically I think it's fixable,  Previewing linked notes popup with pressing CTRL does not work

Better video reference please read the d 3 points up, the awkward pause in the video was me pressing ESC but my keyboard logger was not displaying that 😥.

https://github.com/user-attachments/assets/7d70e3de-f50a-4b8e-a04e-942ec0b8b44a

Obsidian with plugins: I have about 28 obsidian plugins and it works fine for me can't check them all

**Comment by zsviczian** [comment]
There is an Exalidraw specific full-screen mode available on the command palette. That does not close on ESC.

![Image](https://github.com/user-attachments/assets/1ee317bb-bf8e-4b5e-99b0-aafaac5384cc)

**Comment by EaZyStudiio** [comment]
Ohh that glitches out for me just blanks my screen have to close obsidian, to get back to my notes.

**Comment by zsviczian** [comment]
Glad it works! The update part is interesting... since this feature has been there since 2022, nothing has changed on this for the past year or so.
Had it not auto-resolved itself, I would have recommended turning other plugins off to see. I can imagine that some plugins could have an issue with how I transition to full screen.

CTRL+hover works with this full-screen approach.

**Comment by EaZyStudiio** [comment]
It's probably the plugins.

Yes I noticed that the hover preview works now, Changine hotkeys as we speak.

If you don't mind me asking.. but  you did use something like the simulated full screen approach but for mobile, but i noticed you still use the this.contentEl.requestFullscreen() fuction to trigger the browser's native fullscreen API. how were you able to solve the ESC issue 🙏🏽🙏🏽

**Comment by EaZyStudiio** [comment]
BUG Report

> Glad it works! The update part is interesting... since this feature has been there since 2022, nothing has changed on this for the past year or so.

PS. It stopped working again 🙏🏽 pure blank screen

it's not plugins I disable everything even created a new vault...

Phew I figured it out, Why excalidraw Fullscreen toggle blank screen on me it's cause i used... wait after futher testing I  might be going insane Cuz it seems this is an obsidian bug please check out the video it'll better explain

https://github.com/user-attachments/assets/ef4ea6b4-b2ab-467c-a103-5cb477a52d0d



> It's probably the plugins.
> 
> Yes I noticed that the hover preview works now, Changine hotkeys as we speak.
> 
> If you don't mind me asking.. but you did use something like the simulated full screen approach but for mobile, but i noticed you still use the this.contentEl.requestFullscreen() fuction to trigger the browser's native fullscreen API. how were you able to solve the ESC issue 🙏🏽🙏🏽
> 
> Web applications generally don't have the ability to intercept or prevent this fundamental browser behavior for security and user experience reasons.

No worries about this i went through the source code And let's just say only stacked tabs are a problem now 😂
